modules/conversation/service.js

```diff
@@ -0,0 +1,84 @@
+const debug = require('debug')('server:conversation:service');
+const mongoose = require('mongoose');
+const Conversation = require('./model');
+const _ = require('underscore');
+
+class ConversationService {
+    createConversation(title, access, owner) {
+        const conversation = new Conversation({
+            title: title || 'New Conversation',
+            access: access || 'private',
+            owner: owner,
+            moderators: [],
+            members: [owner]
+        });
+
+        debug('Create new Conversation: ' + JSON.stringify(conversation, null, 2));
+
+        return conversation.save();
+    }
+
+    updateConversation(id, properties) {
+        debug('Update Conversation: ' + id);
+        debug('Update Conversation Properties: ' + JSON.stringify(properties, null, 2));
+
+        return Conversation
+            .findById(id)
+            .exec()
+            .then((conversation) => {
+                if (conversation) {
+                    const members = _.union(properties.moderators, properties.members, [conversation.owner]);
+
+                    conversation.title = properties.title || 'New Conversation';
+                    conversation.access = properties.access || 'private';
+                    conversation.moderators = properties.moderators || [];
+                    conversation.members = members || [];
+
```

Comment:
Könnten diese Zuweisungen nicht zu einem Problem führen, wenn man nur einen oder wenige Werte (z.B. nur den Titel) einer Conversation updaten möchte? Dann wären ja die anderen Properties null und die Werte, die schon in der Datenbank für die Conversation stehen, würden durch die hier angegebenen Defaultwerte überschrieben werden.